From 7874b1300b5e6780b3b0dd2203f81bf40cc1e905 Mon Sep 17 00:00:00 2001
From: Jim Kleckner <jim@klekner.net>
Date: Mon, 6 Sep 2021 16:26:52 -0700
Subject: [PATCH] Catch OverflowError

This is an application of the patch to dstat described here:
- https://bugs.launchpad.net/ubuntu/+source/dstat/+bug/1866619/comments/5
---
 dool | 15 +++++++++++----
 1 file changed, 11 insertions(+), 4 deletions(-)

diff --git a/dool b/dool
index 064007d..28d1fc2 100755
--- a/dool
+++ b/dool
@@ -24,6 +24,7 @@ import getopt
 import getpass
 import glob
 import linecache
+import math
 import os
 import re
 import resource
@@ -2108,6 +2109,9 @@ def dchg(var, width, base):
     "Convert decimal to string given base and length"
     c = 0
     while True:
+        if math.isinf(var):
+            ret = 'Inf'
+            break
         ret = str(int(round(var)))
         if len(ret) <= width:
             break
@@ -2242,10 +2246,13 @@ def cprint(var, ctype = 'f', width = 4, scale = 1000):
 
     ### Add unit to output
     if unit:
-        if c != -1 and round(var) != 0:
-            ret += cunit + units[c]
-        else:
-            ret += char['space']
+        try:
+            if c != -1 and round(var) != 0:
+                ret += cunit + units[c]
+            else:
+                ret += char['space']
+        except OverflowError:
+            ret += char['space']
 
     return ret
 
